rwengine/src/ai/CharacterController.cpp

```diff
@@ -10,504 +10,450 @@
+  if (vehicle->vehicle->type == VehicleData::BOAT) {
+    character->enterVehicle(vehicle, seat);
+    return true;
+  }
+
+  if (seat == ANY_SEAT) {
+    // Determine which seat to take.
+    float nearest = std::numeric_limits<float>::max();
+    for (unsigned int s = 1; s < vehicle->info->seats.size(); ++s) {
+      auto entry = vehicle->getSeatEntryPositionWorld(s);
+      float dist = glm::distance(entry, character->getPosition());
+      if (dist < nearest) {
+        seat = s;
+        nearest = dist;
+      }
+    }
+  }
+
+  auto entryDoor = vehicle->getSeatEntryDoor(seat);
+  auto entryPos = vehicle->getSeatEntryPositionWorld(seat);
+  auto entryPosLocal = vehicle->getSeatEntryPosition(seat);
+
+  auto anm_open = character->animations.car_open_lhs;
+  auto anm_enter = character->animations.car_getin_lhs;
+  auto anm_pullout = character->animations.car_pullout_lhs;
+
+  if (entryPosLocal.x > 0.f) {
+    anm_open = character->animations.car_open_rhs;
+    anm_enter = character->animations.car_getin_rhs;
+    anm_pullout = character->animations.car_pullout_rhs;
+  }
+
+  // If there's someone in this seat already, we may have to ask them to leave.
+  auto currentOccupant = static_cast<CharacterObject *>(vehicle->getOccupant(seat));
+
+  bool tryToEnter = false;
+
+  if (entering) {
+    if (character->animator->getAnimation(AnimIndexAction) == anm_open) {
+      if (character->animator->isCompleted(AnimIndexAction)) {
+        tryToEnter = true;
+      } else if (entryDoor && character->animator->getAnimationTime(AnimIndexAction) >= 0.5f) {
+        vehicle->setPartTarget(entryDoor, true, entryDoor->openAngle);
+      } else {
+        // character->setPosition(vehicle->getSeatEntryPosition(seat));
+        character->rotation = vehicle->getRotation();
+      }
+    } else if (character->animator->getAnimation(AnimIndexAction) == anm_pullout) {
+      if (character->animator->isCompleted(AnimIndexAction)) {
+        tryToEnter = true;
+      }
+    } else if (character->animator->getAnimation(AnimIndexAction) == anm_enter) {
+      if (character->animator->isCompleted(AnimIndexAction)) {
+        // VehicleGetIn is over, finish activity
+        return true;
+      }
+    }
+  } else {
+    glm::vec3 targetDirection = entryPos - character->getPosition();
+    targetDirection.z = 0.f;
+
+    float targetDistance = glm::length(targetDirection);
+
+    if (targetDistance <= 0.4f) {
+      entering = true;
+      // Warp character to vehicle orientation
+      character->controller->setMoveDirection({0.f, 0.f, 0.f});
+      character->controller->setRunning(false);
+      character->setHeading(glm::degrees(glm::roll(vehicle->getRotation())));
+
+      // Determine if the door open animation should be skipped.
+      if (entryDoor == nullptr || (entryDoor->constraint != nullptr &&
+                                   glm::abs(entryDoor->constraint->getHingeAngle()) >= 0.6f)) {
```

Comment:
This should have broken at the `||`. However, we can fix these issues manually.
Ideally we'd store more complicated logic in a boolean which also describes what's going on: `bool isDoorOpened = ...`